master/security/adopt-zero-trust.md

```diff
@@ -0,0 +1,314 @@
+
+- **NetworkPolicy** and **GlobalNetworkPolicy** with:
+  - Namespaces
+  - RBAC
+  - Service accounts
+- **HostEndpoints**
+- Security group integration with AWS
+- Calico with application layer policy for Istio
+
+### Concepts
+
+#### The network is always hostile
+
+**Zero Trust Networking** is an approach to network security that is unified by the principle that the network is always assumed to be hostile. This is in direct contrast to perimeter and “segmentation” approaches that focus on separating the world into trusted and untrusted network segments.
+
+Why assume the network is hostile? In many attack scenarios, it is.
+
+- Attackers may compromise “trusted” parts of your network infrastructure: routers, switches, links, etc.
+- Deliberate or accidental misconfiguration can route sensitive traffic over untrusted networks, like the public Internet.
+- Other endpoints on a “trusted” network may be compromised: your application may share a network with thousands of other servers, tens of thousands of other containers, thousands of personal laptops, phones, etc.
+
+Major breaches typically start as a minor compromise of as little as a single component, but attackers then use the network to move laterally toward high value targets: your company’s or customers’ data. In a zone or perimeter model, attackers can move freely inside the perimeter or zone after they have compromised a single endpoint. A Zero Trust Network is resilient to this threat because it enforces strong, cryptographic authentication and access control on each and every network connection.
+
+#### Requirements of a Zero Trust Network
+
+Zero Trust Networks rely on network access controls with specific requirements:
+
+**Requirement 1:** All network connections are subject to enforcement (not just those that cross zone boundaries).
+
+**Requirement 2**: Establishing the identity of a remote endpoint is always based on multiple criteria including strong cryptographic proofs of identity. In particular, network-level identifiers like IP address and port are not sufficient on their own as they can be spoofed by a hostile network.
+
+**Requirement 3**: All expected and allowed network flows are explicitly whitelisted. Any connection not explicitly whitelisted is denied.
+
+**Requirement 4**: Compromised workloads must not be able to circumvent policy enforcement.
+
+**Requirement 5**: Many Zero Trust Networks also rely on encryption of network traffic to prevent disclosure of sensitive data to hostile entities snooping network traffic. This is not an absolute requirement if private data are not exchanged over the network, but to fit the criteria of a Zero Trust Network, encryption must be used on every network connection if it required at all. A Zero Trust Network does not distinguish between trusted and untrusted network links or paths. Also note that even when not using encryption for data privacy, cryptographic proofs of authenticity are still used to establish identity.
```

Comment:
```suggestion
**Requirement 5**: Many Zero Trust Networks also rely on encryption of network traffic to prevent disclosure of sensitive data to hostile entities snooping network traffic. This is not an absolute requirement if private data are not exchanged over the network, but to fit the criteria of a Zero Trust Network, encryption must be used on every network connection if it is required at all. A Zero Trust Network does not distinguish between trusted and untrusted network links or paths. Also note that even when not using encryption for data privacy, cryptographic proofs of authenticity are still used to establish identity.
```